packages/aws-cdk-lib/aws-dynamodb/lib/global-table-base.ts

```diff
@@ -0,0 +1,488 @@
+import { DynamoDBMetrics } from './dynamodb-canned-metrics.generated';
+import * as perms from './perms';
+import { Operation, SystemErrorsForOperationsMetricOptions, OperationsMetricOptions, ITable } from './shared';
+import { IMetric, MathExpression, Metric, MetricOptions, MetricProps } from '../../aws-cloudwatch';
+import { Grant, IGrantable } from '../../aws-iam';
+import { IKey } from '../../aws-kms';
+import { Aws, Resource } from '../../core';
+
+export interface IGlobalTable extends ITable {
+  /**
+   * The ID of the Replica Table in the deployment region of the Global Table.
+   *
+   * @attribute
+   */
+  readonly tableId?: string;
+}
+
+/**
+ * Base class for a Global Table.
+ */
+export abstract class GlobalTableBase extends Resource implements IGlobalTable {
+  /**
+   * The ARN of the Replica Table in the deployment region of the Global Table.
+   *
+   * @attribute
+   */
+  public abstract readonly tableArn: string;
+
+  /**
+   * The name of all Replica Tables in the Global Table.
+   *
+   * @attribute
+   */
+  public abstract readonly tableName: string;
+
+  /**
+   * The stream ARN of the Replica Table in the deployment region of the Global Table.
+   *
+   * @attribute
+   */
+  public abstract readonly tableStreamArn?: string;
+
+  /**
+   * The ID of the Replica Table in the deployment region of the Global Table.
+   *
+   * @attribute
+   */
+  readonly tableId?: string;
+
+  /**
+   * The KMS encryption key associated with the Replica Table in the deployment region
+   * of the Global Table.
+   */
+  public abstract readonly encryptionKey?: IKey;
+
+  protected abstract readonly region: string;
+
+  protected readonly replicaKeys: { [region: string]: IKey } = {};
+
+  protected readonly replicaArns: string[] = [];
+
+  protected readonly streamArns: string[] = [];
+
+  protected abstract get hasIndex(): boolean;
+
+  /**
+   * Adds an IAM policy statement associated with this Global Table to an IAM principal's policy.
+   *
+   * Note: Appropriate grants will also be added to the customer-managed KMS keys associated with this
+   * Global Table and its Replicas.
+   *
+   * @param grantee the principal (no-op if undefined)
+   * @param actions the set of actions to allow (i.e., 'dynamodb:PutItem', 'dynamodb:GetItem', etc.)
+   */
+  public grant(grantee: IGrantable, ...actions: string[]): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        this.tableArn,
+        this.hasIndex ? `${this.tableArn}/index/*` : Aws.NO_VALUE,
+        ...this.replicaArns,
+        ...this.replicaArns.map(arn => this.hasIndex ? `${arn}/index/*` : Aws.NO_VALUE),
```

Comment:
You don't need `Aws.NO_VALUE` here -- you already know `hasIndex` is either true or false, so you can just splat in the strings, or not:

```suggestion
        ...(this.hasIndex ? this.replicaArns.map(arn => `${arn}/index/*`) : [])
```

Reply:
Thanks for pointing this out. This was removed as part of the update to how `grantXxx` is being handled.